### PR TITLE
Refactor FileSystem Get & Put interfaces

### PR DESCRIPTION
**Pull request by rjesquivias**

### CODE REVIEW

File: src/FileSystemReceiver/FileSystemPut.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <string>
+
+class FileSystemPut {
+public:
+    virtual ~FileSystemPut() {}
+
+    virtual bool putFile(std::string) = 0;
+};
```
Comment (crgodfirnon) [inline src/FileSystemReceiver/FileSystemPut.h]: For future reference, when you rename files you should do use git mv to preserve the file history.

File: src/FileSystemReceiver/FileSystemReceiver.h [hunk 2]
```diff
@@ -22,8 +22,8 @@ class FileSystemReceiver : public FileSystemChangeDirectory,
     virtual ReturnTypes listDirectory() = 0;
     virtual ReturnTypes changeDirectory(std::string) = 0;
     virtual ReturnTypes execute(std::string) = 0;
-    virtual bool isFileAtCurrentDirectory() = 0;
-    virtual bool isFileAtGivenDirectory(std::string) = 0;
+    virtual bool getFile() = 0;
+    virtual bool putFile(std::string) = 0;
 
     inline void setDirectory(std::string d) { directory = d; }
     inline std::string getDirectory() { return directory; }
```
Comment (crgodfirnon) [inline src/FileSystemReceiver/FileSystemReceiver.h]: do we care about using the keyword const on functions that are const?

File: src/FileSystemReceiver/WindowsFileSystemReceiver.h [hunk 3]
```diff
@@ -12,8 +12,8 @@ class WindowsFileSystemReceiver : public FileSystemReceiver
 	ReturnTypes listDirectory() { return ReturnTypes::SUCCESS;  }
     ReturnTypes changeDirectory(std::string);
     ReturnTypes execute(std::string) { return ReturnTypes::SUCCESS; }
-    bool isFileAtCurrentDirectory() { return true; }
-    bool isFileAtGivenDirectory(std::string) { return true; }
+    bool getFile() { return true; }
```
Comment (crgodfirnon) [inline src/FileSystemReceiver/WindowsFileSystemReceiver.h]: const

File: src/FileSystemReceiver/FileSystemReceiver.h (hunk 2 above)
Comment (rjesquivias) [reply src/FileSystemReceiver/FileSystemReceiver.h]: Added

**Pull request merged**